nnpdf_data/nnpdf_data/theorydbutils.py

```diff
@@ -51,7 +51,17 @@ def fetch_theory(theory_database: Path, theoryID: int):
     >>> from nnpdf_data.theorydbutils import fetch_theory
     >>> theory = fetch_theory(theory_cards, 700)
     """
-    filepath = theory_database / f"{theoryID}.yaml"
+
+    # Since theoryfile names may contain underscores, the theoryfile name cannot uniquely be
+    # determined from the theoryID. Therefore we create a mapping between the theoryid as python
+    # int and the corresponding file
+    available_theories = {
+        int(theoryfile.name.removesuffix('.yaml')): theoryfile.name
+        for theoryfile in theory_database.glob("*.yaml")
+    }
+    theoryfile = available_theories[theoryID]
```

Comment:
```suggestion
    try:
        theoryfile = available_theories[theoryID]
    except KeyError:
         raise TheoryNotFoundInDatabase(f"Theory {theoryID} not found in database") 
```

(this fixes one of the errors in the tests, not sure about the others)